obsreport/obsreport.go

```diff
@@ -61,9 +33,8 @@ func buildComponentPrefix(componentPrefix, configType string) string {
 	return componentPrefix + configType + obsmetrics.NameSep
 }
 
-func errToStatus(err error) trace.Status {
+func recordError(span trace.Span, err error) {
```

Comment:
In the operator, we have a `handleError(trace.Span, error) error`, which returns the same error it received, adding the error to the span if it's non-nil. The caller can then just do a `return handleError(span, error)`.

Reply:
The caller does not need to return the error, so it is just unnecessary to return an error. But that is a good pattern to follow if needed.